Issue while creating Storage Policy in vCenter Server

### Code of Conduct

- [X] I have read and agree to the [Code of Conduct](https://github.com/vmware/power-validated-solutions-for-cloud-foundation/blob/main/CODE_OF_CONDUCT.md).
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### VMware Cloud Foundation

5.1

### Module Version

2.9.0.1029

### PowerShell Version

7.3.6

### PowerCLI Version

13.1.0.21624340

### PowerVCF Version

2.4.1.1000

### Guest Operating System

windows server 2019

### Environment Details

_No response_

### Description

Issue while running Invoke-DRIDeployment

### Error or Debug Output

[INFO] Creating Storage Policy in vCenter Server ([sfo-w01-vc01.sfo.rainpole.io](http://sfo-w01-vc01.sfo.rainpole.io/)) named (vsphere-with-tanzu-storage-policy): SUCCESSFUL
[ERROR] 3/12/2024 09:15:59 Get-SpbmStoragePolicy SpbmStoragePolicy with name 'vsphere-with-tanzu-storage-policy' was not found using the specified filter(s).
[ERROR] 3/12/2024 09:15:59 Get-SpbmStoragePolicy SpbmStoragePolicy with name 'vsphere-with-tanzu-storage-policy' was not found using the specified filter(s).
[ERROR] 3/12/2024 09:15:59 Get-SpbmStoragePolicy SpbmStoragePolicy with name 'vsphere-with-tanzu-storage-policy' was not found using the specified filter(s).

### Expected Behavior

There should not be any errors

### Actual Behavior

We are getting errors while running Invoke-drideployment function

### Steps to Reproduce

Run Invoke-DriDeployment

### Log Fragments and Files

_No response_

### Screenshots

_No response_

### References

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -49,6 +49,7 @@
 - Fixed `Invoke-GlobalWsaDeployment` cmdlet failure to programmatically resolve the cross-instance Workspace One Access FQDN.
 - Fixed `Invoke-IomDeployment` cmdlet to point out the correct certificate folder.
 - Fixed `Set-LocalUserPasswordExpiration` cmdlet if statement for `warnDays`.
+- Fixed `Invoke-DriDeployment` cmdlet to handle message output for error during execution of `Add-StoragePolicy`.
 - Enhanced `Export-vROpsJsonSpec` cmdlet to support automatic creation of anti-affinity rule for the VMware Aria Operations cluster nodes.
 - Enhanced `Request-vRSLCMBundle` cmdlet to improve the progress tracking.
 - Enhanced `Get-WSAServerDetail` cmdlet to handle single node Workspace ONE Access deployments.

--- a/PowerValidatedSolutions.psd1
+++ b/PowerValidatedSolutions.psd1
@@ -11,7 +11,7 @@
     RootModule        = 'PowerValidatedSolutions.psm1'
 
     # Version number of this module.
-    ModuleVersion     = '2.9.0.1038'
+    ModuleVersion     = '2.9.0.1039'
 
     # Supported PSEditions
     # CompatiblePSEditions = @()

--- a/PowerValidatedSolutions.psm1
+++ b/PowerValidatedSolutions.psm1
@@ -8947,7 +8947,8 @@ Function Invoke-DriDeployment {
                             if (!$failureDetected) {
                                 Show-PowerValidatedSolutionsOutput -message "Creating a Storage Policy that Uses the New vSphere Tag for $solutionName"
                                 $StatusMsg = Add-StoragePolicy -server $jsonInput.sddcManagerFqdn -user $jsonInput.sddcManagerUser -pass $jsonInput.sddcManagerPass -domain $jsonInput.tanzuSddcDomainName -policyName $jsonInput.storagePolicyName -tagName $jsonInput.tagName -WarningAction SilentlyContinue -ErrorAction SilentlyContinue -WarningVariable WarnMsg -ErrorVariable ErrorMsg
-                                messageHandler -statusMessage $StatusMsg -warningMessage $WarnMsg -errorMessage $ErrorMsg; if ($ErrorMsg) {$failureDetected = $true}
+                                if ($StatusMsg -or $WarnMsg) {$null = $ErrorMsg} elseif ($ErrorMsg) {$failureDetected = $true}
+                                messageHandler -statusMessage $StatusMsg -warningMessage $WarnMsg -errorMessage $ErrorMsg
                             }
 
                             if (!$failureDetected) {

